### DISCUSSION TITLE
Does the 'formula' parameter alter state assignment?

### DISCUSSION THREAD

**Discussion by a-u-j** [bmcclintock/momentuHMM#192] [Q&A] [answered]
I understand that the Viterbi algorithm is used to generate the most probable sequence of states from your model. However, if your model contains elements in the 'formula' parameter (e.g. `formula = ~cosinor(hour, period=24)*temperature + vegetation_category + sex`) will this alter how the states are assigned to each datapoint, relative to a model without a formula parameter?

Furthermore, I'm trying to understand where the 'formula' parameter fits more generally in a Hidden Markov model. I.e., if I had run a simple model, without a formula, derived the states from this model, and _then_ used the states as the response variable in a multinomial logistic regression, with those covariates as predictors; would this give me the same outcome as if I just included those covariates in the formula parameter (i.e `formula = ~cosinor(hour, period=24)*temperature + vegetation_category + sex`)? Does including the formula parameter simply remove an extra step in the process, or does it have a further purpose (i.e. affecting the underlying statistics involved in assigning states?)

**Comment by bmcclintock** [comment, marked as answer]
I would expect the differences to be case dependent, but, if covariates are included in the state transition probabilities via the `formula` argument, then they are included in the HMM and the Viterbi algorithm. This is not equivalent to fitting a simple model without covariates and then using the Viterbi-derived states as the response variable in a multinomial logistic regression.
